src/UnscentedTransformKalmanInversion.jl

```diff
@@ -0,0 +1,474 @@
+    uu_cov::Any  # ::Iterable{AbstractMatrix{FT}}
+    "an iterable of arrays of size `N_y` containing the predicted observation (in each `uki` iteration a new array of predicted observation is added)"
+    obs_pred::Any # ::Iterable{AbstractVector{FT}}
+    "weights in UKI"
+    c_weights::Union{AbstractVector{FT}, AbstractMatrix{FT}}
+    mean_weights::AbstractVector{FT}
+    cov_weights::AbstractVector{FT}
+    "number of particles 2N+1 or N+2"
+    N_ens::IT
+    "covariance of the artificial evolution error"
+    Σ_ω::AbstractMatrix{FT}
+    "covariance of the artificial observation error"
+    Σ_ν_scale::FT
+    "regularization parameter"
+    α_reg::FT
+    "regularization vector"
+    r::AbstractVector{FT}
+    "update frequency"
+    update_freq::IT
+    "using augmented system (Tikhonov regularization with Kalman inversion in Chada 
+    et al 2020 and Huang et al (2022)) to regularize the inverse problem, which also imposes prior 
+    for posterior estimation."
+    impose_prior::Bool
+    "prior mean - defaults to initial mean"
+    prior_mean::Any
+    "prior covariance - defaults to initial covariance"
+    prior_cov::Any
+    "current iteration number"
+    iter::IT
+    "used to store matrices: buffer[1] = Y' *Γ_inv, buffer[2] = Y' * Γ_inv * Y"
+    buffer::AbstractVector
+end
+
+function TransformUnscented(
+    u0_mean::VV,
+    uu0_cov::MM;
+    α_reg::FT = 1.0,
+    update_freq::IT = 0,
+    modified_unscented_transform::Bool = true,
+    impose_prior::Bool = false,
+    prior_mean::Any = nothing,
+    prior_cov::Any = nothing,
+    sigma_points::String = "symmetric",
+) where {FT <: AbstractFloat, IT <: Int, VV <: AbstractVector, MM <: AbstractMatrix}
+
+    u0_mean = FT.(u0_mean)
+    uu0_cov = FT.(uu0_cov)
+    if impose_prior
+        if isnothing(prior_mean)
+            @info "`impose_prior=true` but `prior_mean=nothing`, taking initial mean as prior mean."
+            prior_mean = u0_mean
+        else
+            prior_mean = FT.(prior_mean)
+        end
+        if isnothing(prior_cov)
+            @info "`impose_prior=true` but `prior_cov=nothing`, taking initial covariance as prior covariance"
+            prior_cov = uu0_cov
+        else
+            prior_cov = FT.(prior_cov)
+        end
+        α_reg = 1.0
+        update_freq = 1
+    end
+
+    if sigma_points == "symmetric"
+        N_ens = 2 * size(u0_mean, 1) + 1
+    elseif sigma_points == "simplex"
+        N_ens = size(u0_mean, 1) + 2
+    else
+        throw(ArgumentError("sigma_points type is not recognized. Select the option \"symmetric\", \"simplex\"... "))
+    end
+
+    N_par = size(u0_mean, 1)
+    # ensemble size
+
+    mean_weights = zeros(FT, N_ens)
+    cov_weights = zeros(FT, N_ens)
+
+    if sigma_points == "symmetric"
```

Comment:
All of this logic is shared with regular UKI, is it not? It would be good to not duplicate the code if possible.

Reply:
Mostly yes, I have changed it so we now build the `Unscented` object with it's constructor (avoiding duplicates) then we move the fields over to `TransformUnscented`

Reply:
Done